modules/managers/galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager/src/test/java/dev/galasa/zos3270/spi/TerminalTest.java

```diff
@@ -45,5 +50,106 @@ public boolean connectClient() throws NetworkException {
+
+        String terminalId = "abcdef";
+        String host = "myHost";
+        int port = 1234;
+        boolean isSSL = false;
+        boolean isVerifyServer = false;
+        TerminalSize primarySize = new TerminalSize(80,24);
+        TerminalSize alternateSize = new TerminalSize(80,24);
+        ITextScannerManagerSpi textScan = null ;
+        Charset codePage = Charset.availableCharsets().get("1024");
+
+        ByteArrayOutputStream networkOut = new ByteArrayOutputStream();
+
+        MockNetwork network = new MockNetwork(networkOut) {
+            public boolean connectClient() throws NetworkException {
+                return true;
+            }
+        };
+
+        Terminal terminal = new Terminal(terminalId, host, port, isSSL, isVerifyServer, primarySize, alternateSize,
+        textScan, codePage, network);
+
+        String terminalJsonStr = terminal.toJsonString();
+        assertThat(terminalJsonStr).isNull();
+    }
+
+    @Test
+    public void testCanConvertTerminalToJson() throws Exception {
+        GalasaGson gson = new GalasaGson();
+
+        String terminalId = "abcdef";
+        String host = "myHost";
+        int port = 1234;
+        boolean isSSL = false;
+        boolean isVerifyServer = false;
+        TerminalSize primarySize = new TerminalSize(80,24);
+        TerminalSize alternateSize = new TerminalSize(80,24);
+        ITextScannerManagerSpi textScan = null ;
+        Charset codePage = Charset.availableCharsets().get("1024");
+
+        ByteArrayOutputStream networkOut = new ByteArrayOutputStream();
+
+        MockNetwork network = new MockNetwork(networkOut) {
+            public boolean connectClient() throws NetworkException {
+                return true;
+            }
+        };
+
+        Terminal terminal = new Terminal(terminalId, host, port, isSSL, isVerifyServer, primarySize, alternateSize,
+        textScan, codePage, network);
+
+        String textContent = "Hello, world!";
+        Character[] fieldChars = textContent.chars().mapToObj(c -> (char)c).toArray(Character[]::new);
+        String runId = "run1";
+        int sequenceNo = 0;
+        boolean isInbound = true;
+        String type = "test";
+        String aid = "ENTER";
+        int cursorColumn = 0;
+        int cursorRow = 10;
+        dev.galasa.zos3270.common.screens.Terminal rasTerminal = new dev.galasa.zos3270.common.screens.Terminal(terminalId, runId, sequenceNo, primarySize);
+        TerminalImage image = new TerminalImage(0, terminalId, isInbound, type, aid, primarySize, cursorColumn, cursorRow);
+
+        TerminalField field = new TerminalField(0, 0, false, false, false,
+            true, false, false, false, Colour.BLUE.getLetter(),
+                Colour.DEFAULT.getLetter(), Highlight.DEFAULT.getLetter()
+        );
+        FieldContents fieldContents = new FieldContents(fieldChars);
+        field.getContents().add(fieldContents);
+        image.getFields().add(field);
```

Comment:
This might benefit from some comments for those who aren't as familiar with 3270, as I'm not sure what's going on here and why the TerminalField for example is being set up like that, is there a reason or is it arbitrary? Anyway, a few short comments might help here.